packages/manager/apps/pci-quota/src/pages/regions/Regions.page.tsx

```diff
@@ -0,0 +1,169 @@
+import { OdsButton, OdsMessage, OdsText } from '@ovhcloud/ods-components/react';
+import { useEffect, useState } from 'react';
+import { Translation, useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import {
+  Notifications,
+  useNotifications,
+} from '@ovh-ux/manager-react-components';
+import {
+  isDiscoveryProject,
+  PciAnnouncementBanner,
+  PciDiscoveryBanner,
+  useProject,
+} from '@ovh-ux/manager-pci-common';
+import { useMedia } from 'react-use';
+import { AvailablePart } from '@/pages/regions/Available.part';
+import { ToAddPart } from '@/pages/regions/ToAdd.part';
+import { useGetAvailableRegions, useLocations } from '@/api/hooks/useRegions';
+import { addRegion } from '@/api/data/region';
+import queryClient from '@/queryClient';
+
+export type TPlainLocation = {
+  name: string;
+  continent: string;
+  regions: string[];
+};
+
+type TState = {
+  isAddingRegion: boolean;
+};
+
+export default function RegionsPage(): JSX.Element {
+  const { t } = useTranslation('regions');
+
+  const [state, setState] = useState<TState>({ isAddingRegion: false });
+
+  const isMobile: boolean = useMedia(`(max-width: 760px)`);
+
+  const { addSuccess, addError, clearNotifications } = useNotifications();
+
+  const { projectId } = useParams();
+  const { data: project } = useProject();
+
+  const { data: availableRegions } = useGetAvailableRegions(projectId);
+
+  const { data: locations, isPending } = useLocations(projectId, true);
+
+  const [selectedLocation, setSelectedLocation] = useState<TPlainLocation>(
+    undefined,
+  );
+
+  const [selectedRegions, setSelectedRegions] = useState<
+    Record<string, string>
+  >({});
+
+  const setSelectedRegion = (location: string, region: string) => {
+    setSelectedRegions((prev) => ({ ...prev, [location]: region }));
+  };
+
+  useEffect(() => {
+    locations?.forEach((location) => {
+      setSelectedRegion(location.name, location.regions[0]);
+    });
+  }, [locations]);
+
+  const doAddRegion = async () => {
+    setState({ ...state, isAddingRegion: true });
+    try {
+      const code = selectedRegions[selectedLocation.name];
+      await addRegion(projectId, code);
+      queryClient.setQueryData(
+        ['project', projectId, 'availableRegions'],
+        availableRegions.filter((region) => region.name !== code),
+      );
```

Comment:
Please invalidate query call instead and factorize query key to avoid differing implementation